pipeline_dp/private_spark.py

```diff
@@ -12,10 +13,32 @@ class PrivateRDD:
     within the specified privacy budget can be extracted from it through its API.
     """
 
-    def __init__(self, rdd, budget_accountant, privacy_id_extractor):
-        self._rdd = rdd
+    def __init__(self, rdd, budget_accountant, privacy_id_extractor=None):
+        if privacy_id_extractor:
+            self._rdd = rdd.map(lambda x: (privacy_id_extractor(x), x))
+        else:
+            self._rdd = rdd
         self._budget_accountant = budget_accountant
-        self._privacy_id_extractor = privacy_id_extractor
+
+    def map(self, fn: Callable) -> PrivateRDD:
+        """ A Spark mapValues equivalent.
+
+        Keeps the state (namely budget_accountant and privacy_id_extractor).
```

Comment:
Let's rewrite in the following way:

**Keeps track of privacy_id for each element. The output PrivateRDD has the same BudgetAccountant as self.**

Reply:
changed